sources/packages/backend/apps/queue-consumers/src/processors/schedulers/esdc-integration/ecert-integration/_tests_/ecert-full-time-process-integration.scheduler.e2e-spec.ts

```diff
@@ -401,6 +411,303 @@ describe(
+        const fakeCanadaLoanOverawardBalance = createFakeDisbursementOveraward({
+          student,
+        });
+        fakeCanadaLoanOverawardBalance.disbursementValueCode = "CSLF";
+        fakeCanadaLoanOverawardBalance.overawardValue = -550;
+        // Create fake overaward for BCSL.
+        // The value is smaller than the deducted amount in the original assessment
+        // to ensure the credit will be applied only partially.
+        const fakeBCLoanOverawardBalance = createFakeDisbursementOveraward({
+          student,
+        });
+        fakeBCLoanOverawardBalance.disbursementValueCode = "BCSL";
+        fakeBCLoanOverawardBalance.overawardValue = -700;
+        await db.disbursementOveraward.save([
+          fakeCanadaLoanOverawardBalance,
+          fakeBCLoanOverawardBalance,
+        ]);
+
+        // Queued job.
+        const mockedJob = mockBullJob<void>();
+
+        // Act
+        await processor.processQueue(mockedJob.job);
+
+        // Assert some log messages related to overaward processing.
+        expect(
+          mockedJob.containLogMessages([
+            "Checking if overaward adjustments are needed.",
+            "Found overaward balance for the student.",
+            "Applying overaward credit for CSLF.",
+            "Applying overaward credit for BCSL.",
+          ]),
+        ).toBe(true);
+        // Assert overawards were credited.
+        const [firstSchedule] =
+          application.currentAssessment.disbursementSchedules;
+        const overawards = await db.disbursementOveraward.find({
+          select: {
+            id: true,
+            overawardValue: true,
+            disbursementValueCode: true,
+            originType: true,
+          },
+          where: {
+            disbursementSchedule: { id: firstSchedule.id },
+          },
+          order: { disbursementValueCode: "ASC" },
+        });
+        expect(overawards.length).toBe(2);
+        const [bcslOveraward, cslfOveraward] = overawards;
+        expect(cslfOveraward).toEqual({
+          id: expect.any(Number),
+          overawardValue: 500,
+          disbursementValueCode: "CSLF",
+          originType: DisbursementOverawardOriginType.AwardCredited,
+        });
+        expect(bcslOveraward).toEqual({
+          id: expect.any(Number),
+          overawardValue: 700,
+          disbursementValueCode: "BCSL",
+          originType: DisbursementOverawardOriginType.AwardCredited,
+        });
+        // Assert awards
+        // Select all awards generated for the schedule.
+        const awards = await db.disbursementValue.find({
+          where: { disbursementSchedule: { id: firstSchedule.id } },
+        });
+        // Assert CSLF.
+        const hasExpectedCSLF = awards.filter(
+          (award) =>
+            award.valueCode === "CSLF" &&
+            award.disbursedAmountSubtracted === 5000 &&
+            award.overawardAmountSubtracted === -500 &&
+            award.effectiveAmount === 500,
+        );
```

Comment:
I am not against this way of assertion. But we also assert the whole object with selected properties and can we follow that way?